pymc3/model.py

```diff
@@ -1695,7 +1695,13 @@ def pandas_to_array(data):
     XXX: When `data` is a generator, this will return a Theano tensor!
 
     """
-    if hasattr(data, "values"):  # pandas
+    if isinstance(data, np.ndarray):
+        mask = np.isnan(data)
+        if np.any(mask):
+            ret = np.ma.MaskedArray(data, mask)
+        else:
+            ret = data
+    elif hasattr(data, "values"):  # pandas
         if data.isnull().any().any():  # missing values
```

Comment:
While we are here, should we also avoid the repetition `data.isnull()` in the pandas branch?